test/standalone/Soil/soiltest.jl

```diff
@@ -169,10 +169,10 @@ for FT in (Float32, Float64)
         set_initial_cache! = make_set_initial_cache(soil_heat_on)
         set_initial_cache!(p, Y, t0)
 
-        # Test explicit tendency - should update internal energy, set others to 0
-        exp_tendency! = make_exp_tendency(soil_heat_on)
+        # Test implicit tendency - should update internal energy only (Ksat = 0 for hydrology)
```

Comment:
It might be nice to add a comment at the top of this test explaining that water and energy are stepped implicitly so we expect `imp_tendency` to update those, and `exp_tendency` to update ice. The only downside is we'd have to remember to update that comment if we change what's stepped implicitly vs explicitly

Reply:
I added this!